inject/src/main/java/io/micronaut/inject/annotation/AnnotationMetadataHierarchy.java

```diff
@@ -955,4 +1028,48 @@ public Optional<String> findRepeatableAnnotation(String annotation) {
         }
         return Optional.empty();
     }
+
+    /**
+     * Merges the hierarchy into one {@link MutableAnnotationMetadata}.
+     *
+     * @return merged metadata
+     * @since 4.0.0
+     */
+    public MutableAnnotationMetadata merge() {
+        MutableAnnotationMetadata newAnnotationMetadata = new MutableAnnotationMetadata();
+        for (AnnotationMetadata annotationMetadata : hierarchy) {
```

Comment:
I think you would need to loop in reverse order because the first entry in the hierarchy is the highest priority. Can you add a test for this?

Reply:
It works this way because the values aren't overridden when added. Test added